pygama/dsp/_processors/fixed_time_pickoff.py

```diff
@@ -1,20 +1,44 @@
 import numpy as np
 from numba import guvectorize
+import math
+from pygama.dsp.errors import DSPFatal
 
-@guvectorize(["void(float32[:], float64, float32[:])",
-              "void(float64[:], float64, float64[:])"],
+
+@guvectorize(["void(float32[:], float32, float32[:])",
+              "void(float64[:], float32, float64[:])"],
              "(n),()->()", nopython=True, cache=True)
-def fixed_time_pickoff(wf_in, time, value_out):
+
+def fixed_time_pickoff(w_in, t_in, a_out):
     """
-    Return the waveform value at the requested time. Interpolate between
-    samples for floating point times.
+    Fixed time pickoff -- gives the waveform value at a fixed time
+    Parameters
+    ----------
+    w_in : array-like
+            Input waveform
+    t_in : float
+            Time point to find value
+    a_out : float
+            Output value
+    Processing Chain Example
+    ------------------------
+    "trapEftp": {
+        "function": "fixed_time_pickoff",
+        "module": "pygama.dsp.processors",
+        "args": ["wf_trap", "tp_0 + (10*us+2.5*us)", "trapEftp"],
+        "unit": "ADC",
+        "prereqs": ["wf_trap", "tp_0"]
+        },
     """
-    if not (time >= 0 and time <= len(wf_in-1)):
-        value_out[0] = np.nan
     
-    t = int(time)
-    frac = time-t
-    if frac==0:
-        value_out[0] = wf_in[t]
-    else:
-        value_out[0] = wf_in[t]*(1-frac) + wf_in[t+1]*frac
+    a_out[0] = np.nan
+
+    if (np.isnan(w_in).any() or np.isnan(t_in)):
+        return
+    
+    if (not int(t_in) in range(len(w_in))):
+        return
+  
+    if (not np.floor(t_in)==t_in):
```

Comment:
This line is throwing errors for me, since the time_point_finder returns a float

Reply:
I've removed this condition as it also causes problems when using the values obtained by the optimizer